src/lsd.cpp

```diff
@@ -242,7 +242,7 @@ void lsd::on_announce(udp::endpoint const& from, char const* buf
 	{
 		// we expect it to be hexadecimal
 		// if it isn't, it's not our cookie anyway
-		std::int32_t const cookie = strtol(cookie_iter->second.c_str(), nullptr, 16);
+		std::int32_t const cookie = int(strtol(cookie_iter->second.c_str(), nullptr, 16));
```

Comment:
yes, this is OK, creation is `m_cookie((random(0x7fffffff) ^ std::uintptr_t(this)) & 0x7fffffff)`